service/raft/raft_group0.cc

```diff
@@ -218,15 +227,15 @@ future<> raft_group0::join_group0() {
     if (!_raft_gr.is_enabled() || std::holds_alternative<raft::group_id>(_group0)) {
         co_return;
     }
-    auto my_addr = co_await load_or_create_my_addr();
-    raft::group_id group0_id = raft::group_id{co_await db::system_keyspace::get_raft_group0_id()};
-    if (group0_id != raft::group_id{}) {
-        rslog.trace("{} is starting group 0 with id {}", my_addr.id, group0_id);
-        co_await _raft_gr.start_server_for_group(create_server_for_group(group0_id, my_addr));
-        _group0 = group0_id;
-        co_return;
+
+    auto group0_id = raft::group_id{co_await db::system_keyspace::get_raft_group0_id()};
+    if (group0_id) {
```

Comment:
Honestly, it really annoys me when little tweaks like these happen. I think the previous version was more declarative.:
raft::group_id group0_id = raft::group_id{co_await db::system_keyspace::get_raft_group0_id()};
    if (group0_id != raft::group_id{}) {
    
    We didn't use auto. - it's easier to see what object type is if there is no auto. You could argue that there is raft::group0_id on the right, but what is the justification for the change? 
    The next line as well. When you compare with an empty object you don't have to quess what operator bool or operator! are doing in the class. It was more explicit. 
    
    I don't mind if someone writes their code in less explicit manner, but going over someone else's code (mine in this case) and tweaking it like that... ist just ... well.. annoying.

Reply:
> You could argue that there is raft::group0_id on the right, but what is the justification for the change?

Well - why not? What's the point of writing out the type twice? It annoyed me so I changed it :P
I suggest we don't write Java 7:
`HashMap<String, Integer> hashMap = new HashMap<String, Integer>();`

> When you compare with an empty object you don't have to quess what operator bool or operator! are doing in the class. It was more explicit.

Maybe - but then maybe it's a mistake that we have an `operator bool` inside the `group_id` class in the first place if we don't use it?

> I don't mind if someone writes their code in less explicit manner, but going over someone else's code (mine in this case) and tweaking it like that... ist just ... well.. annoying.

Sorry :(